File: src/app/services/reaction.service.ts
import {Injectable} from '@angular/core';
import {ApiService} from './api.service';
import {ConfigService} from './config.service';
import { Reaction } from '../models/reaction';



@Injectable()
export class ReactionService {

  constructor(
    private apiService: ApiService,
    private config: ConfigService
  ) {}
  
  
  reactToPost(postId: number, reaction: Reaction){
   const url = `${this.config.reaction_url}/reactToPost/${postId}`;
    return this.apiService.post(url, reaction);
  }
  findReactionByPostAndUser(postId : number, userId : number) {
    return this.apiService.get(this.config.reaction_url + "/find/post/" + postId +  "/user/"  + userId );
   }

   countReactionsByPost(postId : number) {
    return this.apiService.get(this.config.reaction_url + "/count/post/" + postId );
   }
   reactToComment(commentId: number, reaction: Reaction){
    const url = `${this.config.reaction_url}/reactToComment/${commentId}`;
     return this.apiService.post(url, reaction);
   }
   findReactionByCommentAndUser(commentId : number, userId : number) {
    return this.apiService.get(this.config.reaction_url + "/find/comment/" + commentId + "/user/" + userId );
   }

   countReactionsByComment(commentId : number) {
    return this.apiService.get(this.config.reaction_url + "/count/comment/" + commentId );
   }
    }

File: socialNetwork/src/main/java/com/ftn/socialNetwork/service/intefraces/ReactionService.java
package com.ftn.socialNetwork.service.intefraces;

import com.ftn.socialNetwork.model.entity.*;

import java.util.Map;

public interface ReactionService {

  Reaction create(Reaction reaction);

  Reaction findReactionByPostAndUser(Post post, User user);

  Reaction findReactionByCommentAndUser(Comment comment, User user);

  Map<EReactionType, Integer> countReactionsByPost(Post post);

  Map<EReactionType, Integer> countReactionsByComment(Comment comment);
}
